lib/gems/pending/util/postgres_admin.rb

```diff
@@ -111,18 +111,39 @@ def self.restore(opts)
   end
 
   def self.before_restore(opts)
+    # Drop subscriptions, unload extension and ensure pglogical connections are closed before proceeding
+    unload_pglogical_extension(opts)
+  rescue AwesomeSpawn::CommandResultError
+    $log.info("MIQ(#{name}.#{__method__}) Ignoring failure to remove pglogical before restore ...")
+  end
+
+  def self.unload_pglogical_extension(opts)
     runcmd("psql", opts, :command => <<-SQL)
       SELECT
         drop_subscription
       FROM
         pglogical.subscription subs,
         LATERAL pglogical.drop_subscription(subs.sub_name)
     SQL
+
     runcmd("psql", opts, :command => <<-SQL)
-      DROP EXTENSTION pglogical CASCADE
+      DROP EXTENSION pglogical CASCADE
     SQL
-  rescue AwesomeSpawn::CommandResultError
-    $log.info("MIQ(#{name}.#{__method__}) Ignoring failure to remove pglogical before restore ...")
+
+    # Wait for pglogical manager connection to quiesce. Bail after 5 minutes
```

Comment:
@jrafanie I probably wouldn't ask the user to retry, but I wouldn't object to mirroring the log message to `STDOUT`.